src/operation/decision_model.jl

```diff
@@ -4,47 +4,46 @@ Default PowerSimulations Operation Problem Type
 
 """
-    DecisionModel(::Type{M},
-    template::ProblemTemplate,
-    sys::PSY.System,
-    jump_model::Union{Nothing, JuMP.Model}=nothing;
-    kwargs...) where {M<:DecisionProblem,
-                      T<:PM.AbstractPowerFormulation}
+    DecisionModel{M}(
+        template::ProblemTemplate,
+        sys::PSY.System,
+        jump_model::Union{Nothing, JuMP.Model}=nothing;
+        kwargs...) where {M<:DecisionProblem}
 
 This builds the optimization problem of type M with the specific system and template.
 
 # Arguments
 
   - `::Type{M} where M<:DecisionProblem`: The abstract operation model type
-  - `template::ProblemTemplate`: The model reference made up of transmission, devices,
-    branches, and services.
+  - `template::ProblemTemplate`: The model reference made up of transmission, devices, branches, and services.
   - `sys::PSY.System`: the system created using Power Systems
   - `jump_model::Union{Nothing, JuMP.Model}`: Enables passing a custom JuMP model. Use with care
-
-# Output
-
-  - `model::DecisionModel`: The operation model containing the model type, built JuMP model, Power
-    Systems system.
+  - `name = nothing`: name of model, string or symbol; defaults to the type of template converted to a symbol.
+  - `optimizer::Union{Nothing,MOI.OptimizerWithAttributes} = nothing` : The optimizer does
+    not get serialized. Callers should pass whatever they passed to the original problem.
+  - `horizon::Int = UNSET_HORIZON`: Manually specify the length of the forecast Horizon
+  - `warm_start::Bool = true`: True will use the current operation point in the system to initialize variable values. False initializes all variables to zero. Default is true
+  - `system_to_file::Bool = true:`: True to create a copy of the system used in the model.
+  - `initialize_model::Bool = true`: Option to decide to initialize the model or not.
+  - `initialization_file::String = ""`: TODO
+  - `deserialize_initial_conditions::Bool = false`: Option to deserialize conditions
+  - `export_pwl_vars::Bool = false`: True to export all the pwl intermediate variables. It can slow down significantly the solve time.
+  - `allow_fails::Bool = false`: True to allow the simulation to continue even if the optimization step fails. Use with care.
+  - `optimizer_solve_log_print::Bool = false`: Uses JuMP.unset_silent() to print the optimizer's log. By default all solvers are set to MOI.Silent()
+  - `detailed_optimizer_stats::Bool = false`: True to save detailed optimizer stats log.
+  - `calculate_conflict::Bool = false`: True to use solver to calculate conflicts for infeasible problems. Only specific solvers are able to calculate conflicts.
+  - `direct_mode_optimizer::Bool = false`: True to use the solver in direct mode. Creates a [JuMP.direct_model](https://jump.dev/JuMP.jl/dev/reference/models/#JuMP.direct_model).
+  - `store_variable_names::Bool = false`: True to store variable names in optimization model.
+  - `rebuild_model::Bool = false`: TODO
```

Comment:
```suggestion
  - `rebuild_model::Bool = false`: It will force the rebuild of the underlying JuMP model with each call to update the model. It increases solution times, use only if the model can't be updated in memory. 
```